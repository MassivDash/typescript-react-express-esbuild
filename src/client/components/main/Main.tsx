import React, { ReactElement } from 'react';

export interface MainProps {
  children: ReactElement;
}

const Main: React.FunctionComponent<MainProps> = ({ children }) => {
  return (
    <main className="mt-10 mx-auto max-w-7xl px-4 sm:mt-12 sm:px-6 md:mt-16 lg:mt-20 lg:px-8 xl:mt-28">
      {children}
    </main>
  );
};

export default Main;
