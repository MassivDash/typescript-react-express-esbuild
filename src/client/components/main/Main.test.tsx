import { render } from '@testing-library/react';
import React from 'react';
import Main, { MainProps } from './Main';

describe('Main', () => {
  const defaultProps: MainProps = {
    children: <p>Spaceghost</p>,
  };

  it('should render', () => {
    const props = { ...defaultProps };
    const { asFragment, queryByText } = render(<Main {...props} />);

    expect(asFragment()).toMatchSnapshot();
    expect(queryByText('Spaceghost')).toBeTruthy();
  });
});
