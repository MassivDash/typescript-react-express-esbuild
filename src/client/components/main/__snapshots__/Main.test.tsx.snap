// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render 1`] = `
<DocumentFragment>
  <main
    class="mt-10 mx-auto max-w-7xl px-4 sm:mt-12 sm:px-6 md:mt-16 lg:mt-20 lg:px-8 xl:mt-28"
  >
    <p>
      Spaceghost
    </p>
  </main>
</DocumentFragment>
`;
